### PR TITLE
Fusion: Prompt reset scene context on saving to another task

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Add a prompt to reset comp or content in scene values to the folder or task being saved to.

## Additional info

image

I'll be preparing a similar matching feature for Houdini and Maya.

## Testing notes:

1. Open a comp
2. With workfiles tool, save it to another folder or task
3. The prompt should appear and be clear what it does.

### CODE REVIEW

**Comment by BigRoy** [comment]
@EmberLightVFX is this something you might be interested in testing?

**Comment by EmberLightVFX** [comment]
> @EmberLightVFX is this something you might be interested in testing?

When I get some time over I'll do!
Just a question. What does "Reset" mean in this context? Would it mean the checked items would get their info from the current folder/task or the folder/task it's being copied over to?

**Comment by BigRoy** [comment]
> Just a question. What does "Reset" mean in this context? Would it mean the checked items would get their info from the current folder/task or the folder/task it's being copied over to?

It will set it to the folder/task you're saving into. So that if you save your comp to e.g. another folder you directly have it set up to match e.g. that shots' frame range (if you picked that option).

File: client/ayon_core/hosts/fusion/api/lib.py [hunk 1]
```diff
@@ -285,3 +308,99 @@ def comp_lock_and_undo_chunk(
...
+    from ayon_core.lib import BoolDef, UILabelDef
+    from qtpy import QtWidgets, QtCore
+
+    definitions = [
+        UILabelDef(
+            label=(
+                "You are saving your scene into a different task."
+                "\n\n"
+                "Would you like to reset some settings for the "
+                "for the new context?\n"
```
Comment (EmberLightVFX) [inline client/ayon_core/hosts/fusion/api/lib.py:353]: You have two "for the" here.

**Comment by EmberLightVFX** [comment]
> It will set it to the folder/task you're saving into. So that if you save your comp to e.g. another folder you directly have it set up to match e.g. that shots' frame range (if you picked that option).

Maybe it's just me that finds the sentence `Would you like to reset some settings for the new context?` as it's not my first language but what do you think about `Would you like to change some settings with the new context?`

**Comment by BigRoy** [comment]
> > It will set it to the folder/task you're saving into. So that if you save your comp to e.g. another folder you directly have it set up to match e.g. that shots' frame range (if you picked that option).
> 
> Maybe it's just me that finds it a bit hard to understand `Would you like to reset some settings for the new context?` as it's not my first language but what do you think about `Would you like to change some settings with the new context?`

Thanks, I've updated it and tweaked the wording.

image

If you think that looks great and is clear - then I'll match it in the Houdini and Maya PRs that match this feature.

File: client/ayon_core/hosts/fusion/api/lib.py (hunk 1 above)
Comment (BigRoy) [reply client/ayon_core/hosts/fusion/api/lib.py]: Fixed with https://github.com/ynput/ayon-core/pull/259/commits/a94081e820f562dd7574607a4ce4e24bada666a5

**Comment by EmberLightVFX** [comment]
The only thing I see is the title of the window says "saving to different context" while the first row says "...saving your scene to a different task".

Ifk exactly how the terminaligy is for Ayon but should "scene" be "product" and "task" be "context"?

**Comment by BigRoy** [comment]
> Ifk exactly how the terminaligy is for Ayon but should "scene" be "product" and "task" be "context"?

I think scene should then be `workfile` - and `context` is basically the "folder path + task".

I'll change:
- `scene` -> `workfile`
- `task` -> `context`

**Comment by BigRoy** [comment]
image

Actually - I tried one more different thing.

By mentioning that you are saving to a different "folder" or "task" I'm hoping this clarifies a bit what the 'context' is the pop-up refers to. What do you think of this?

**Comment by EmberLightVFX** [comment]
> By mentioning that you are saving to a different "folder" or "task" I'm hoping this clarifies a bit what the 'context' is the pop-up refers to. What do you think of this?

Love it!

**Comment by BigRoy** [comment]
I fixed the merge conflicts - should be ready to merge.

**Pull request merged**